_cffi_build/dice_one_against_many.cpp

```diff
@@ -465,8 +465,20 @@ extern "C"
         node_queue top_k_scores(score_cmp(), std::move(vec));
 
         uint32_t count_one = _popcount_array(comp1, keywords);
-        if (count_one == 0)
-            return 0;
+        if (count_one == 0) {
+            if (threshold > 0) {
+                return 0;
+            }
+
+            for (uint32_t j = 0; j < k; ++j) {
+                scores[j] = 0.0;
+            }
+            for (uint32_t j = 0; j < k; ++j) {
+                indices[j] = j;
+            }
+
```

Comment:
It would look more natural (read: require fewer lines) to write this as
```
if (threshold > 0)
    return 0;
for (uint32_t j = 0; j < k; ++j) {
    scores[j] = 0.0;
    indices[j] = j;
}
```

Reply:
Yes, that’s a more natural way of doing things.

I split it into two loops because the former seems very easy to vectorise, whereas the latter looks harder. I thought that this would help the compiler perform vectorisation. But I don’t actually know much about this. Do you have any thoughts?

Reply:
Good point! I’ve simplified it.